### ISSUE TITLE
Where is the 20231214 update?

### ISSUE THREAD

**Issue by Firestar-Reimu** [outloudvi/mw2fcitx#13]
I own an AUR repo https://aur.archlinux.org/packages/fcitx5-moegirl

and I am thinking of updating it once a month

but December update has not released yet.

**Comment by Firestar-Reimu** [comment]
https://github.com/outloudvi/mw2fcitx/actions/runs/7214341037
```
[build-and-publish](https://github.com/outloudvi/mw2fcitx/actions/runs/7214341037/job/19656214732#step:3:535)
Process completed with exit code 1.

[build-and-publish](https://github.com/outloudvi/mw2fcitx/actions/runs/7214341037/job/19656214732)
The following actions uses node12 which is deprecated and will be forced to run on node16: actions/checkout@v2. For more info: https://github.blog/changelog/2023-06-13-github-actions-all-actions-will-run-on-node16-instead-of-node12-by-default/
```

**Comment by outloudvi** [comment]
Seems that some 404 occurred. Probably a breaking API change on MediaWiki versions. I'll look into it some time later.

Thanks for reporting!

**Comment by outloudvi** [comment]
Seems that Moegirlpedia disabled an API we need to list page names.

https://zh.moegirl.org.cn/api.php?action=query&list=allpages

```json
{
    "error": {
        "code": "action-notallowed",
        "info": "\u29fcapierror-action-notallowed\u29fd",
        "*": "See https://zh.moegirl.org.cn/api.php for API usage. Subscribe to the mediawiki-api-announce mailing list at &lt;https://lists.wikimedia.org/mailman/listinfo/mediawiki-api-announce&gt; for notice of API deprecations and breaking changes."
    }
}
```

**Comment by Firestar-Reimu** [comment]
So this pinyin dictionary will never be updated ever since?

**Comment by outloudvi** [comment]
It depends. I'll try to contact Moegirlpedia on the API change, but I think the change is by design and thus they are unlikely to revert the API restriction. Despite that, there are still alternative solutions like using one of the (somehow) mirrors of MGP.

**Comment by outloudvi** [comment]
I [got a reply](https://zh.moegirl.org.cn/Special:Diff/7256427) from Moegirlpedia admin that they are doing some server side changes. In the future, they will set up another endpoint dedicated for bots, which may hopefully be helpful to this project. Before that, the repo might not publish updates of *-pinyin-moegirl.

**Comment by outloudvi** [comment]
@tumuyan Yep. We can get through [Special:NewPages](https://zh.moegirl.org.cn/Special:NewPages) to have a list of new titles, and the list's also [possible with the MediaWiki API](https://en.wikipedia.org/wiki/Special:ApiSandbox#action=query&format=json&list=recentchanges&formatversion=2&rctype=new). However, it's still not solving this very issue, since Moegirlpedia doesn't open access to this API.
The list of new titles might not be too long (I think, <2000 to make up for 2 months), but I don't think the situation is so tense that the project has to bypass the API and do the selenium trick.

**Comment by outloudvi** [comment]
[A new version](https://github.com/outloudvi/mw2fcitx/releases/tag/20240309) is published by looking at `Special:NewPages`. We believe that several requests to `Special:NewPages` should not generate too much load, and thus decided to do it. Thanks, @tumuyan!

**Comment by outloudvi** [comment]
Closing as the API method is [now working](https://github.com/outloudvi/mw2fcitx/actions/runs/16855487096/job/47747804407) since Moegirlpedia re-enabled API access after upgrading their MediaWiki instance version.
